src/steps/create_and_checkout_branch_step.go

```diff
@@ -15,5 +15,7 @@ type CreateAndCheckoutBranchStep struct {
 // Run executes this step.
 func (step *CreateAndCheckoutBranchStep) Run() error {
 	git.SetParentBranch(step.BranchName, step.ParentBranchName)
-	return script.RunCommand("git", "checkout", "-b", step.BranchName, step.ParentBranchName)
+	err := script.RunCommand("git", "checkout", "-b", step.BranchName, step.ParentBranchName)
+	git.UpdateCurrentBranchCache(step.BranchName)
+	return err
```

Comment:
The way this is written right now it could be simplified to:

```go
git.UpdateCurrentBranchCache(step.BranchName)
return script.RunCommand("git", "checkout", "-b", step.BranchName, step.ParentBranchName)
```
Did you intend to check the `err` variable in your code here?

Reply:
It needs to be after `script.RunCommand` as otherwise the branch name the script is run on  will be wrong. I'll add a comment explaining that

Reply:
Ended up just wrapping them in a block checking that err is nil